contracts/compound/Comptroller.sol

```diff
@@ -464,8 +499,39 @@ contract Comptroller is ComptrollerV3Storage, ComptrollerInterface, ComptrollerE
       return uint256(Error.SUPPLIER_NOT_WHITELISTED);
     }
 
-    // Check borrow cap
-    uint256 borrowCap = borrowCaps[cToken];
+    PrudentiaConfig memory capConfig = borrowCapConfig;
+
+    uint256 borrowCap;
+
+    // Check if we're using Adrastia Prudentia for the borrow cap
+    if (capConfig.controller != address(0)) {
+      // We have a controller, so we're using Adrastia Prudentia
+
+      // Get the borrow cap from Adrastia Prudentia
+      borrowCap = IHistoricalRates(capConfig.controller).getRateAt(cToken, capConfig.offset).current;
+
+      // Prudentia stores the borrow cap as a whole token amount while our code requires the mantissa amount, so we
+      // must scale the borrow cap by the underlying token decimals
+
+      uint256 underlyingDecimals = 18;
+
+      // Try to get the underlying asset address (we use a staticcall b/c CEther doesn't have an underlying)
+      (bool success, bytes memory data) = cToken.staticcall(abi.encodeWithSignature("underlying()"));
```

Comment:
after the refactoring from #638 is merged, the CEther/CToken will removed from the codebase in favour of CErc20, so these calls can be simplified

Reply:
Thanks for the speedy review, Veliko! I've simplified the calls for getting the underlying token address.